lib/teleterm/gateway/gateway.go

```diff
@@ -214,13 +216,20 @@ func (g *Gateway) LocalPortInt() int {
 }
 
 // CLICommand returns a command which launches a CLI client pointed at the given gateway.
+// It needs to return a relative command as it will be executed from a terminal in Connect, so we
+// should let user's env resolve the command rather than depending on os.exec.
 func (g *Gateway) CLICommand() (string, error) {
-	cliCommand, err := g.cfg.CLICommandProvider.GetCommand(g)
```

Comment:
I realized that the code would be way easier to test and have more sound design if `CLICommandProvider.GetCommand` just returns an `os.exec.Cmd` and then `CLICommand` later just takes stuff from it.

This way in tests I can return a "mocked" `Cmd` which simulates a situation where a relative command gets resolved to an absolute path. Otherwise I wouldn't be able work my way around `exec.Command` and would need to reintroduce its mocked version again.